irohad/execution/query_execution_impl.hpp

```diff
@@ -0,0 +1,120 @@
+
+    bool validate(const shared_model::interface::Query &query,
+                  const shared_model::interface::GetRolePermissions
+                      &get_role_permissions);
+
+    bool validate(
+        const shared_model::interface::Query &query,
+        const shared_model::interface::GetAccountAssets &get_account_assets);
+
+    bool validate(const shared_model::interface::Query &query,
+                  const shared_model::interface::GetAccount &get_account);
+
+    bool validate(
+        const shared_model::interface::Query &query,
+        const shared_model::interface::GetSignatories &get_signatories);
+
+    bool validate(const shared_model::interface::Query &query,
+                  const shared_model::interface::GetAccountTransactions
+                      &get_account_transactions);
+
+    bool validate(const shared_model::interface::Query &query,
+                  const shared_model::interface::GetAccountAssetTransactions
+                      &get_account_asset_transactions);
+
+    bool validate(
+        const shared_model::interface::Query &query,
+        const shared_model::interface::GetAccountDetail &get_account_detail);
+
+    bool validate(
+        const shared_model::interface::Query &query,
+        const shared_model::interface::GetTransactions &get_transactions);
+
+    QueryResponseBuilderDone executeGetAssetInfo(
+        const shared_model::interface::GetAssetInfo &get_asset_info);
+
+    QueryResponseBuilderDone executeGetRoles(
+        const shared_model::interface::GetRoles &query);
+
+    QueryResponseBuilderDone executeGetRolePermissions(
+        const shared_model::interface::GetRolePermissions &query);
+
+    QueryResponseBuilderDone executeGetAccountAssets(
+        const shared_model::interface::GetAccountAssets &query);
+
+    QueryResponseBuilderDone executeGetAccountDetail(
+        const shared_model::interface::GetAccountDetail &query);
+
+    QueryResponseBuilderDone executeGetAccount(
+        const shared_model::interface::GetAccount &query);
+
+    QueryResponseBuilderDone executeGetSignatories(
+        const shared_model::interface::GetSignatories &query);
+
+    QueryResponseBuilderDone executeGetAccountAssetTransactions(
+        const shared_model::interface::GetAccountAssetTransactions &query);
+
+    QueryResponseBuilderDone executeGetAccountTransactions(
+        const shared_model::interface::GetAccountTransactions &query);
+
+    QueryResponseBuilderDone executeGetTransactions(
+        const shared_model::interface::GetTransactions &query,
+        const shared_model::interface::types::AccountIdType &accountId);
+
+    QueryResponseBuilderDone executeGetPendingTransactions(
+        const shared_model::interface::GetPendingTransactions &query,
+        const shared_model::interface::types::AccountIdType &query_creator);
+
+    std::shared_ptr<ametsuchi::WsvQuery> wsvQuery_;
+    std::shared_ptr<ametsuchi::BlockQuery> blockQuery_;
+    std::mutex m;
```

Comment:
One of the reasons to introduce connection pool and SOCI was to remove locks. Please remove the mutex and use a unique connection for each query from the client.